content/cosign-and-policy-controller-with-gke-artifact-registry-and-cloud-kms.md

```diff
@@ -129,52 +129,67 @@ The following checks were performed on each of these signatures:
-
 Enable the GKE API in our current project:
 ```bash
 gcloud services enable container.googleapis.com
 ```
 
-Create a GKE cluster with this dedicated GSA:
+Create a GKE cluster with Workload Identity:
 ```bash
 gcloud container clusters create ${CLUSTER_NAME} \
-    --service-account ${GSA_ID} \
-    --region ${REGION} \
+    --workload-pool=${PROJECT_ID}.svc.id.goog \
+    --zone ${ZONE} \
     --scopes "gke-default,https://www.googleapis.com/auth/cloudkms"
 ```
-_Note: we explicitly add the `https://www.googleapis.com/auth/cloudkms` scope needed by Sistore’s `policy-controller`. [`https://www.googleapis.com/auth/cloud-platform`](https://cloud.google.com/kubernetes-engine/docs/how-to/access-scopes) instead is fine too._
+_Note: we explicitly add the `https://www.googleapis.com/auth/cloudkms` scope needed by Policy-controller. [`https://www.googleapis.com/auth/cloud-platform`](https://cloud.google.com/kubernetes-engine/docs/how-to/access-scopes) instead is fine too._
+
+Define a least privilege Google Service Account (GSA) for Policy-controller by granting the `cloudkms.viewer`, `cloudkms.verifier` and `artifactregistry.reader` roles and by enabling Workload Identity between the Kubernetes Service Account and the Google Service Account:
+```bash
+PC_GSA_NAME=policy-controller-sa
+PC_GSA_ID=${PC_GSA_NAME}@${PROJECT_ID}.iam.gserviceaccount.com
+
+gcloud iam service-accounts create ${PC_GSA_NAME}
+
+gcloud iam service-accounts add-iam-policy-binding ${PC_GSA_ID} \
+    --role roles/iam.workloadIdentityUser \
+    --member "serviceAccount:${PROJECT_ID}.svc.id.goog[cosign-system/policy-controller-policy-webhook]"
+gcloud iam service-accounts add-iam-policy-binding ${PC_GSA_ID} \
+    --role roles/iam.workloadIdentityUser \
+    --member "serviceAccount:${PROJECT_ID}.svc.id.goog[cosign-system/policy-controller-webhook]"
+
+gcloud projects add-iam-policy-binding ${PROJECT_ID} \
+    --role roles/cloudkms.verifier \
+    --member serviceAccount:${PC_GSA_ID}
+gcloud projects add-iam-policy-binding ${PROJECT_ID} \
+    --role roles/cloudkms.viewer \
+    --member serviceAccount:${PC_GSA_ID}
+
+gcloud artifacts repositories add-iam-policy-binding ${REGISTRY_NAME} \
+    --location ${REGION} \
+    --member "serviceAccount:${PC_GSA_ID}" \
+    --role roles/artifactregistry.reader
+```
 
-Install the [Sigstore’s `policy-controller` Helm chart](https://github.com/sigstore/helm-charts/tree/main/charts/policy-controller) in this GKE cluster:
+Install the [Policy-controller Helm chart](https://github.com/sigstore/helm-charts/tree/main/charts/policy-controller) in this GKE cluster by annotating the Policy-controller's `ServiceAccounts` to use Workload Identity:
 ```bash
 helm repo add sigstore https://sigstore.github.io/helm-charts
 helm repo update
-helm install policy-controller \
-    -n cosign-system sigstore/policy-controller \
-    --create-namespace
+helm upgrade policy-controller \
```

Comment:
Is this the same as 'apply', meaning you can upgrade from 'nothing' :)

Reply:
Yep, good question, `helm upgrade --install` will do the same as `helm install` for the first time.

Reply:
Yes, I'd prefer to use install here too.